config/src/parameters/actual.rs

```diff
@@ -71,16 +71,24 @@ impl Root {
 pub struct Common {
     pub chain_id: ChainId,
     pub key_pair: KeyPair,
-    pub p2p_address: SocketAddr,
+    pub p2p: P2p,
 }
 
 impl Common {
     /// Construct an id of this peer
     pub fn peer_id(&self) -> PeerId {
-        PeerId::new(self.p2p_address.clone(), self.key_pair.public_key().clone())
+        PeerId::new(self.p2p.address.clone(), self.key_pair.public_key().clone())
     }
 }
 
+/// Common options shared between multiple places
+#[allow(missing_docs)]
+#[derive(Debug, Clone)]
+pub struct P2p {
+    pub address: SocketAddr,
+    pub idle_timeout: Duration,
+}
```

Comment:
```suggestion
/// Network options
#[allow(missing_docs)]
#[derive(Debug, Clone)]
pub struct Network {
    pub address: SocketAddr,
    pub idle_timeout: Duration,
}
```

Reply:
Initially i used Network name, problem is that it clashes with other Network structs defined through codebase. 
What is our policy here? rename to Config?

Reply:
I would say, the approach I used is to reduce discrepancy between config modules and actual Iroha core modules. That is why `iroha_config::parameters::actual` even exists - its purpose is to map _user_ configuration in a way that lies down onto the internal structure.

I would appreciate if you take it one step further in this direction. Maybe it would mean to make a minor refactoring on the core side as well. Maybe it would mean to get rid of `actual::Sumeragi` and define just `actual::Network`. I am confused myself.

Reply:
i don't think we should merge p2p configuration and sumeragi configuration as single entity, they even defined in separate crates

Reply:
what we can do is to duplicate some fields instead of having `Common` struct

Reply:
renamed `P2p` to `Network` and moved it to the `Root` also `Common` now has `PeerId` directly

Reply:
@0x009922 your opinion?

Reply:
I think it is a fine compromise, thanks.